### PR TITLE
Add ENV proxy in Dockerfile

### PR DESCRIPTION
**Pull request by bhavanijayakumaran**
2. Check for docker install and do not fail just print the message, startdocker will start the docker daemon if docker is not running..

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,5 +1,9 @@
 FROM debian:11
 
+# Add the proxy to get around the proxy issues with m1 mac mini with apt-get commands
```
[the remark this answers is not in the document]
Comment (riftEmber) [reply Dockerfile:3]: This is likely a problem to be solved on the host machine, or the Docker install of the host machine, or on the network it is on (but I know nothing about that).

File: util/cron/docker.bash [hunk 2]
```diff
@@ -8,7 +8,7 @@ check_docker_install(){
     else
         echo "ERROR! Docker Desktop not installed:"
         echo "  * Install docker desktop from <https://docs.docker.com/docker-for-mac/install/>"
-        exit 1
```
[the remark this answers is not in the document]
Comment (bhavanijayakumaran) [reply util/cron/docker.bash:11]: We just check and install docker in the script. Failing here just exits the test run which don't want.

File: Dockerfile (hunk 1 above)
[the remark this answers is not in the document]
Comment (bhavanijayakumaran) [reply Dockerfile:3]: I tried to setup proxy in the host machine, Since there is an issue in apt-get debian package during image build, this ENV is required inside the Dockerfile.

[the remark this answers is not in the document]
Comment (riftEmber) [reply Dockerfile:3]: Let me rephrase my objection as a question -- what about the problem makes it so that it needs to be solved in the Dockerfile as opposed to anywhere else?

[the remark this answers is not in the document]
Comment (bhavanijayakumaran) [reply Dockerfile:3]: apt-get commands are failing with the below 403 errors.

Adding the proxy to the etc/environment file in does not take effect in docker file.


---> Running in d9da40616e9a
Err:1 http://deb.debian.org/debian bullseye InRelease
403 Forbidden [IP: 151.101.58.132 80]
Err:2 http://deb.debian.org/debian-security bullseye-security InRelease
403 Forbidden [IP: 151.101.58.132 80]
Err:3 http://deb.debian.org/debian bullseye-updates InRelease
403 Forbidden [IP: 151.101.58.132 80]
Reading package lists...
�[91mE: The repository 'http://deb.debian.org/debian bullseye InRelease' is not signed.
E: Failed to fetch http://deb.debian.org/debian/dists/bullseye/InRelease 403 Forbidden [IP: 151.101.58.132 80]
E: Failed to fetch http://deb.debian.org/debian-security/dists/bullseye-security/InRelease 403 Forbidden [IP: 151.101.58.132 80]
E: The repository 'http://deb.debian.org/debian-security bullseye-security InRelease' is not signed.
E: Failed to fetch http://deb.debian.org/debian/dists/bullseye-updates/InRelease 403 Forbidden [IP: 151.101.58.132 80]
E: The repository 'http://deb.debian.org/debian bullseye-updates InRelease' is not signed.

[the remark this answers is not in the document]
Comment (bhavanijayakumaran) [reply Dockerfile:3]: Just curious, Is anyone else using dockerfile to build chapel images other than our team?

[the remark this answers is not in the document]
Comment (riftEmber) [reply Dockerfile:3]: I do not know what you're referring to by etc/environment file as machine configuration is out of scope of my knowledge, but that is not an answer to why this problem *cannot be solved outside of the Dockerfile*.

One of my suggestions was perhaps this needs to be fixed in the Docker install on the host machine. I googled "docker build use proxy" and the [first result](https://docs.docker.com/network/proxy/) looks like what you need.

[the remark this answers is not in the document]
Comment (bhavanijayakumaran) [reply Dockerfile]: Ok, I added the below in the host machine(that will work if we change docker desktop to something other than colima docker) in addition docker.json(this is specific to colima docker)
Restarting docker daemon fixed the proxy issue.
 Docker proxy:
Create a file /etc/systemd/system/docker.service.d/http-proxy.conf
[Service]
Environment="HTTP_PROXY=http://web-proxy.houston.hpecorp.net:8080" "HTTPS_PROXY=http://web-proxy.houston.hpecorp.net:8080" "NO_PROXY=hp.com,hpe.com,hpecorp.net,hpqcorp.net,localhost,local host,127.0.0.1"
Docker DNS settings:
Create a file /etc/docker/daemon.json as ROOT
{
  "dns": ["16.110.135.52", "16.110.135.51" ]
}

**Review by riftEmber — APPROVED** [review]
